url7b90c1010b_DastanIqbal_HelloZMQwithJava_tgz-pJ8-ClientJ8/src/zmq/ZMQ.java

```diff
@@ -0,0 +1,814 @@
+    {
+        switch (option_) {
+            case ZMQ_MORE:
+                return msg_.has_more() ? 1: 0;
+            default:
+                throw new IllegalArgumentException();
+        }
+    }
+
+    public static void zmq_sleep(int s) {
+        try {
+            Thread.sleep(s*(1000L));
+        } catch (InterruptedException e) {
+        }
+        
+    }
+
+    //  The proxy functionality
+    public static boolean zmq_proxy (SocketBase frontend_, SocketBase backend_, SocketBase control_)
+    {
+        if (frontend_ == null || backend_ == null) {
+            ZError.errno (ZError.EFAULT);
+            throw new IllegalArgumentException();
+        }
+        return Proxy.proxy (
+            frontend_,
+            backend_,
+            control_);
+    }
+
+    @Deprecated
+    public static boolean zmq_device (int device_, SocketBase insocket_,
+            SocketBase outsocket_)
+    {
+        return Proxy.proxy(insocket_, outsocket_, null);
+    }
+
+    /**
+     * Polling on items. This has very poor performance.
+     * Try to use zmq_poll with selector
+     * CAUTION: This could be affected by jdk epoll bug
+     *
+     * @param items_
+     * @param timeout_
+     * @return number of events
+     */
+    public static int zmq_poll(PollItem[] items_, long timeout_) {
+        return zmq_poll(items_, items_.length, timeout_);
+    }
+
+    /**
+     * Polling on items. This has very poor performance.
+     * Try to use zmq_poll with selector
+     * CAUTION: This could be affected by jdk epoll bug
+     *
+     * @param items_
+     * @param timeout_
+     * @return number of events
+     */
+    public static int zmq_poll(PollItem[] items_, int count, long timeout_) {
+        Selector selector = null;
+        try {
+            selector = PollSelector.open();
+        } catch (IOException e) {
+            throw new ZError.IOException(e);
+        }
+
+        int ret = zmq_poll (selector, items_, count, timeout_);
+
+        //  Do not close selector
+        
+        return ret;
+    }
+
+    /**
+     * Polling on items with given selector
+     * CAUTION: This could be affected by jdk epoll bug
+     *
+     * @param selector Open and reuse this selector and do not forget to close when it is not used.
+     * @param items_
+     * @param timeout_
+     * @return number of events
+     */
+    public static int zmq_poll(Selector selector, PollItem[] items_, long timeout_)
+    {
+        return zmq_poll(selector, items_, items_.length, timeout_);
+    }
+    /**
+     * Polling on items with given selector
+     * CAUTION: This could be affected by jdk epoll bug
+     *
+     * @param selector Open and reuse this selector and do not forget to close when it is not used.
+     * @param items_
+     * @param count
+     * @param timeout_
+     * @return number of events
+     */
+    public static int zmq_poll(Selector selector, PollItem[] items_, int count, long timeout_)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 30. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 624-632 into a separate method.